daemon/client_test.go

```diff
@@ -164,6 +171,69 @@ func TestListProcsForUnauthorizedUser(t *testing.T) {
 	assert.Equal(t, "Unauthorized Access!!!\nPlease check the EMAIL_ID and ACCESS_TOKEN validity in proctor config file.", err.Error())
 }
 
+func TestListProcsReturnProcListIfClientVersionHttpHeaderIsEmpty(t *testing.T) {
+	body := `[ { "name": "job-1", "description": "job description", "image_name": "hub.docker.com/job-1:latest", "env_vars": { "secrets": [ { "name": "SECRET1", "description": "Base64 encoded secret for authentication." } ], "args": [ { "name": "ARG1", "description": "Argument name" } ] } } ]`
+	proctorConfig := config.ProctorConfig{Host: "proctor.example.com", Email: "proctor@example.com", AccessToken: "access-token"}
+	proctorClient := NewClient(proctorConfig)
+	var args = []env.VarMetadata{env.VarMetadata{Name: "ARG1", Description: "Argument name"}}
+	var secrets = []env.VarMetadata{env.VarMetadata{Name: "SECRET1", Description: "Base64 encoded secret for authentication."}}
+	envVars := env.Vars{Secrets: secrets, Args: args}
+	var procListExpected = []proc.Metadata{proc.Metadata{Name: "job-1", Description: "job description", EnvVars: envVars}}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterStubRequest(
+		httpmock.NewStubRequest(
+			"GET",
+			"http://"+proctorConfig.Host+"/jobs/metadata",
+			func(req *http.Request) (*http.Response, error) {
+				return httpmock.NewStringResponse(200, body), nil
+			},
+		).WithHeader(
+			&http.Header{
+				utility.UserEmailHeaderKey:   []string{"proctor@example.com"},
+				utility.AccessTokenHeaderKey: []string{"access-token"},
+			},
+		),
+	)
+
+	procList, err := proctorClient.ListProcs()
+
+	assert.Equal(t, procListExpected, procList)
+	assert.NoError(t,err)
+}
+
+func TestListProcsReturnProcListBadrequestError(t *testing.T) {
+	proctorConfig := config.ProctorConfig{Host: "proctor.example.com", Email: "proctor@example.com", AccessToken: "access-token"}
+	proctorClient := NewClient(proctorConfig)
+	var procListExpected = []proc.Metadata{}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterStubRequest(
+		httpmock.NewStubRequest(
+			"GET",
+			"http://"+proctorConfig.Host+"/jobs/metadata",
+			func(req *http.Request) (*http.Response, error) {
+				return httpmock.NewStringResponse(400, "You are proctor client version 0.1.0 outdated. Please upgrade to latest proctor client to continue use proctor!"), nil
```

Comment:
Pull this out as a string since it's used in 2 places.
Grammar correction: `fmt.Sprintf("Your Proctor client is using an outdated version: %s . To continue using proctor, please upgrade to latest version.", version.ClientVersion)`
Put client version in error message since that's what proctord will put.